### PR TITLE
Moves oldviro symptoms and transformation viruses to current virus system

### PR DESCRIPTION
**Pull request by gurfan**
#I noticed over the past few weeks ago that all of the transformation viruses from roburgers, mommi spaghetti, ect. weren't working for some reason. Rather than just fix this issue I decided to move the whole thing to our current virus system and make some of the symptoms more interesting.

Transformation Viruses are stored in a global list that's created roundstart, so feeding roburgers to multiple people will all give them all copies the same disease. Since these viruses are part of our current virus system, the symptoms can all be extracted, modified, and spliced onto different viruses. 
All of the global transformation viruses have an infection chance of 0%, meaning that the only way to get infected them is through eating whatever gives that virus (nanites, wendigo meat, xenomicrobes, ect.) or through ingesting blood infected with that virus.

Nanites (Roburgers), Autism Nanites (MoMMI Spaghetti), Xenomicrobes, and Wendigo Meat now use the current virus system for their transformations. Previously these all gave you some flavortext messages and maybe some damage before transforming you. Now, the viruses have four stages with the actual transformation being the final stage. **None of the transformation virus symptoms can appear through random mutations, the only way to obtain them is through the virus themself.**

### New Transformation Viruses

**Nanite Disease (Cyborg Transformation)**

Stage 1: 
Neural Rewiring
"Rearranges nerve tissue throughout the body, causing the infected to feel more robotic."

Stage 2:
Oleum Syndrome
"Causes the infected to internally synthesize oil and other inorganic material."

Stage 3:
Metallica Syndrome
"Rapidly replaces some organic tissue in the body, causing limbs and other organs to become robotic."

Stage 4:
Silicus Syndrome
"Rapidly replaces the infected's tissue with inorganic matter, causing them to transform into a cyborg."

**Autism Nanite Disease (MoMMI Spaghetti)**

Stage 1:
Major Neural Rewiring
"Rearranges nerve tissue throughout the body, causing the infected to feel more robotic... and autistic."

Stage 2:
Dysplasia Syndrome
"Rapidly restructures the body of the infected, causing them to shrink in size."

Stage 3:
Supermatter Syndrome
"Causes the infected to experience engineering-related hallucinations."

Stage 4:
Autismus Syndrome
"Rapidly replaces the infected's tissue with inorganic matter. This particular strain seems to cause severe autism in the infected as well."

**Xenomicrobe Disease**

Stage 1:
Foreign-Impulse Syndrome
"Alters the thinking patterns of the infected, causing them to be more aggressive and hostile."

Stage 2:
Mega Laryngitis
"Warps the vocal cords of the infected, resulting in normal speech turning into incoherent hisses. In addition, the infected loses the ability to understand normal language."

Stage 3:
Plasmatic Adaptation
"Induces heavy mutation and optimization in the infected's cellular structure. The infected gains several physical abilities and an affinity for plasma."

Stage 4:
Ripley Syndrome
"Causes the infected to mutate into an alien creature."

**Wendigo Curse**

Stage 1:
Fullness Syndrome
"An unsual symptom that causes the infected to feel hungry, even after eating."

Stage 2:
Gastrointestinal Inflammation
"Inflames the GI tract of the infected, causing relentless vomitting."

Stage 3:
Eldritch Mind Syndrome
"UNKNOWN"

Stage 4:
Curse of the Wendigo
"UNKNOWN"

### Old Viruses (Now Symptoms)

In addition, the following old viruses have returned as symptoms:

**Toxic Compensation**
- Stage 3
- Danger 1
- Rapidly converts brute and burn damage to toxin damage.

**Space Retrovirus (Now called Retrotransposis)**
- Stage 4
- Danger 4
- Stores the UI+UE of the first person infected with the virus, all future infected individuals will have their UI+UE overriden with the stored DNA.

Cockatrice Petrification is now handled by a proc instead of a legacy disease as well.

:cl:
 * bugfix: Roburgers, MoMMI Spaghetti, ect. should all transform people again.
 * rscadd: Reworked transformation viruses to use the modern virus system. They now have their own unique symptoms as well.
  * rscadd: Space Retrovirus and Toxic Compensation are available again as stage 4 and stage 3 symptoms, respectively.

### CODE REVIEW

File: code/game/objects/items/weapons/storage/uplink_kits.dm [hunk 1]
```diff
@@ -368,3 +368,18 @@ obj/item/weapon/storage/box/syndie_kit/cheaptide
...
+/obj/item/weapon/storage/box/syndie_kit/transformviruses/New()
+	..()
+	var/obj/item/weapon/reagent_containers/glass/bottle/B1 = new /obj/item/weapon/reagent_containers/glass/bottle(src)
+	var/obj/item/weapon/reagent_containers/glass/bottle/B2 = new /obj/item/weapon/reagent_containers/glass/bottle(src)
+	var/obj/item/weapon/reagent_containers/glass/bottle/B3 = new /obj/item/weapon/reagent_containers/glass/bottle(src)
+	new /obj/item/weapon/reagent_containers/food/snacks/meat/wendigo(src)
+
+	B1.reagents.add_reagent(NANITES, 1)
+	B2.reagents.add_reagent(AUTISTNANITES, 1)
+	B3.reagents.add_reagent(XENOMICROBES, 1)
```
Comment (jknpj) [inline code/game/objects/items/weapons/storage/uplink_kits.dm]: unatomic

File: code/modules/virus2/effect/stage_3.dm [hunk 2]
```diff
@@ -605,3 +605,201 @@ datum/disease2/effect/lubefoot/deactivate(var/mob/living/mob)
...
+
+	var/list/valid_internal_organs = list()
+	for(var/datum/organ/internal/I in H.internal_organs)
+		if(I.name != "brain" && !I.robotic)
+			valid_internal_organs += I
+
+	if(prob(75) || valid_external_organs.len) 
+		
+		var/datum/organ/external/E = pick(valid_external_organs)
+		E.status |= ORGAN_ROBOT
```
Comment (jknpj) [inline code/modules/virus2/effect/stage_3.dm]: there's a proc for this, might as well use it.

File: code/game/objects/items/weapons/storage/uplink_kits.dm (hunk 1 above)
Comment (gurfan) [reply code/game/objects/items/weapons/storage/uplink_kits.dm]: removed the bundle

**Comment by PrimeDSS13** [comment]
Retrovirus returning seems fun, good work.

File: code/modules/virus2/predefined.dm [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+
+
+
+var/list/global_diseases = list()
+
+
+
+/proc/create_global_diseases()
+	for(var/datum/disease2/disease/predefined/disease_type in /datum/disease2/disease/predefined)
```
Comment (DamianX) [inline code/modules/virus2/predefined.dm]: this does not work at all, which makes me wonder how much testing these changes received

Comment (gurfan) [reply code/modules/virus2/predefined.dm]: I tested it multiple times including just now and it works fine. Drinking nanites/microbes/whatever gives whoever drank the reagent the global disease created by this proc

Comment (gurfan) [reply code/modules/virus2/predefined.dm]: I suppose I should be using subtypesof instead of this loop, though

Comment (DamianX) [reply code/modules/virus2/predefined.dm]: I really don't see how that's possible: this loop does nothing, global_disease is only supposed to be written to by this proc, and it clearly doesn't

**Comment by DamianX** [comment]
Do you know what was new-ing the thing?

**Comment by gurfan** [comment]
I don't know exactly what it was, I think it was getcopy() being called in update_global_log. I moved the list creation out of New() and into create_global_diseases so the list doesn't get modified more than once.

**Pull request merged**